solarkraft/src/fetcher/callDecoder.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * @license
+ * [Apache-2.0](https://github.com/freespek/solarkraft/blob/main/LICENSE)
+ */
+/**
+ * The code to extract Stellar operations that correspond to Soroban contract calls.
+ *
+ * Igor Konnov, 2024.
+ */
+
+import sdk, { Address } from '@stellar/stellar-sdk'
+import { ContractCallEntry, FieldsMap } from './storage.js'
+import { Maybe, just, none } from '@sweet-monads/maybe'
+import { OrderedMap } from 'immutable'
+
+/**
+ * Decode a contract call from a Horizon operation.
+ * @param op an operation
+ * @param matcher a quick matcher over the contractId to avoid expensive deserialization
+ */
+export async function extractContractCall(
+    op: any,
```

Comment:
Is there no way to be more precise here?

Reply:
I am not actually sure. Stellar SDK has [Operation](https://stellar.github.io/js-stellar-sdk/js-stellar-base_src_operation.js.html), but it does not look like what Horizon API is returning in Stellar SDK. Some of the methods return `sdk.Horizon.ServerApi.OperationRecord` but it's not exactly compatible with what I need.